pupepat/surface.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+
+
+class SurfaceFitter(object):
+
+    def _get_powers(self, degree):
+        """
+        Make a list of exponents for the polynomial terms
+        :param degree: Maximum of the sum of the x and y powers
+        :return: num_exponents, list of x,y powers to use for the polynomial
+        """
+        x_powers, y_powers = np.meshgrid(range(degree + 1), range(degree + 1))
+        powers_to_use = (x_powers + y_powers <= degree)
+        return powers_to_use.sum(), zip(x_powers[powers_to_use], y_powers[powers_to_use])
+
+    def fit(self, x, y, z, degree):
```

Comment:
Since x and y are used in all the methods, maybe they can be instance variables? Otherwise this class probably doesn't need to be a class.

Reply:
Yeah. Realistically this does not need to be a class.